src/inmanta/module.py

```diff
@@ -276,7 +320,176 @@ def get_file_for_version(self, repo: str, tag: str, file: str) -> str:
+        elif install:
+            return self.install(project, module_spec)
+        else:
+            return None
+
+    @abstractmethod
+    def install(self, project: "Project", module_spec: List[InmantaModuleRequirement]) -> Optional[TModule]:
+        """
+        Attempt to install a module given a module spec.
+
+        :param project: The project associated with the module.
+        :param module_spec: The module specification including any constraints on its version.
+        """
+        raise NotImplementedError("Abstract method")
+
+    def path_for(self, name: str) -> Optional[str]:
+        """
+        Returns the path to the module root directory. Should be called prior to configuring the module finder for v1 modules.
+        """
+        raise NotImplementedError("Abstract method")
+
+    @classmethod
+    @abstractmethod
+    def from_path(cls, project: Optional["Project"], path: str) -> TModule:
+        raise NotImplementedError("Abstract method")
+
+    def _get_module_name(self, module_spec: List[InmantaModuleRequirement]) -> str:
+        module_names: Set[str] = {req.project_name for req in module_spec}
+        module_name: str = more_itertools.one(
+            module_names,
+            too_short=ValueError("module_spec should contain at least one requirement"),
+            too_long=ValueError("module_spec should contain requirements for exactly one module"),
+        )
+        return module_name
+
+
+class ModuleV2Source(ModuleSource["ModuleV2"]):
+    def __init__(self, urls: List[str]) -> None:
+        self.urls: List[str] = [url if not os.path.exists(url) else os.path.abspath(url) for url in urls]
+        env.process_env.init_namespace(const.PLUGINS_PACKAGE)
+
+    @classmethod
+    def get_python_package_name(cls, module_name: str) -> str:
+        return ModuleV2.PKG_NAME_PREFIX + module_name.replace("_", "-")
+
+    @classmethod
+    def get_inmanta_module_name(cls, python_package_name: str) -> str:
+        if not python_package_name.startswith(ModuleV2.PKG_NAME_PREFIX):
+            raise ValueError(f"Invalid python package name: should start with {ModuleV2.PKG_NAME_PREFIX}")
+        result: str = python_package_name[len(ModuleV2.PKG_NAME_PREFIX) :].replace("-", "_")
+        if not result:
+            raise ValueError("Invalid python package name: empty module name part.")
+        return result
+
+    @classmethod
+    def get_python_package_requirement(cls, module_req: InmantaModuleRequirement) -> Requirement:
+        return Requirement.parse(
+            str(module_req).replace(module_req.project_name, cls.get_python_package_name(module_req.project_name))
+        )
+
+    @classmethod
+    def get_namespace_package_name(cls, module_name: str) -> str:
+        return f"{const.PLUGINS_PACKAGE}.{module_name}"
+
+    def install(self, project: Optional["Project"], module_spec: List[InmantaModuleRequirement]) -> Optional["ModuleV2"]:
+        env.process_env.init_namespace(const.PLUGINS_PACKAGE)
```

Comment:
Why is this required, isn't the namespace package part of this project?

Reply:
I think it can indeed be removed since it's called at `ModuleSourceV2` construction.

Reply:
done